### DISCUSSION TITLE
Streaming Response

### DISCUSSION THREAD

**Discussion by mekza** [fastapi/fastapi#7121] [Questions] [answered]
Hi,

I am trying to port some `aiohttp` code to `FastAPI`. 

I have a simple (redis) pub-sub system with a producer and consumers. A client connect to a http endpoint then is subscribeb to a channel finally when a producer publishes a message in this channel it gets sent to the connected client as part of the response's body.

aiohttp provides  [StreamResponse](https://docs.aiohttp.org/en/stable/web_reference.html#streamresponse) you can use it the following way:

```python
async def stream_handler(request):
    response = web.StreamResponse()
    await response.prepare(request)
    (channel,) = await redis.subscribe("foobar")
    while True:
        message = await channel.get()
        await response.write(message)
```

Below is how I tried to port that to FastAPI

```Python
from fastapi import FastAPI
from fastapi.responses import StreamingResponse
import aioredis

app = FastAPI()

async def messages_from_redis(channel):
        while True:
            message = await channel.get()
            if message:
                yield message
                print(message)

@app.get("/")
async def main():
    redis = await aioredis.create_redis_pool("redis://localhost:6379/")
    (channel,) = await redis.subscribe("foobar")
    return StreamingResponse(messages_from_redis(channel))

```
I am not even sure I am using the `StreamingResponse` class the right way. It would be really appreciated if someone could help me.

**Comment by Mause** [comment]
That usage looks correct - is it not working as you expect?

**Comment by mekza** [comment]
Sorry I forgot to say that the client gets what it is the pubsub channel only on disconnect

**Comment by Mause** [comment, marked as answer]
And how are you consuming the streamed data?

**Comment by mekza** [comment]
@Mause omg! Thanks for the hint. My http client was buffering...

**Comment by tiangolo** [comment]
Thanks for the help @Mause! 🚀 

Thanks for coming back to close the issue @mekza 🍰 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
